### PR TITLE
feat: add ‍‍‍`spinner‍‍`

### PR DESCRIPTION
**Pull request by mahdieabdollahian**
close: https://github.com/Tap30/web-components/issues/29

### CODE REVIEW

**Comment by amir78729** [comment]
Please add jsdoc to your component (same as `avatar`)

File: src/spinner/spinner.style.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    stroke-linecap: round;
+  }
+
+  .rotating {
+    transform-origin: 300px 300px;
+    animation : rotate 1s linear infinite;
+  }
+
+  .primary {
+    color: var(--tap-sys-color-surface-black);
```
Comment (amir78729) [inline src/spinner/spinner.style.ts:21]: You can add a token for this field and use `--tap-sys-color-surface-black` as a default value:

```suggestion
    color: var(--tap-spinner-color-primary, var(--tap-sys-color-surface-black));
```

With this approach, the user of the component can override the color of the spinner.

File: src/spinner/spinner.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { LitElement, html } from "lit";
+import { property } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+
+export class spinner extends LitElement {
```
Comment (amir78729) [inline src/spinner/spinner.ts:5]: Please define your class name upper case:

```suggestion
export class Spinner extends LitElement {
```

File: src/spinner/index.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { customElement } from "lit/decorators.js";
+import { spinner } from "./spinner";
+import styles from "./spinner.style";
+
+@customElement("tap-spinner")
+export class TapSpinner extends spinner {
```
Comment (amir78729) [inline src/spinner/index.ts:6]:
```suggestion
export class TapSpinner extends Spinner {
```

File: src/spinner/spinner.style.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { css } from "lit";
+
+export default css`
+  :host {
+    box-sizing: border-box;
+  }
+
+  .spinner {
+    width: 20px;
```
Comment (amir78729) [inline src/spinner/spinner.style.ts:9]: Please use tokens and default values (same as the comment for line 21 in this file)

File: src/spinner/spinner.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+				<rect shape-rendering="crispEdges" x="0" y="0" width="300" height="600" fill="url(#Gradient1)"/>
+				<rect shape-rendering="crispEdges"  x="300" y="0" width="300" height="600" fill="url(#Gradient2)"/>
+			</g>
+			</pattern>
+		</defs>
+		<path class="rotating" style="stroke: url(#Pattern);" fill='transparent' stroke-width='80' d='M 364 58 A 250 250 0 1 1 235 58'/>
+	</svg>
+		`;
+	}
+}
```
Comment (amir78729) [inline src/spinner/spinner.ts:38]: Please add a new empty line to this line.

File: src/spinner/spinner.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { classMap } from "lit/directives/class-map.js";
+
+export class spinner extends LitElement {
+	@property() variant:
+		| "primary"
+		| "inverse" = "primary";
+
+	render() {
+		return html`
+		<svg class=${classMap({
```
Comment (amir78729) [inline src/spinner/spinner.ts:12]: Please add `part`.

File: src/spinner/spinner.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+			</linearGradient>
+			<pattern id="Pattern" x="0" y="0" width="600" height="600" patternUnits="userSpaceOnUse">
+			<g>
+				<rect shape-rendering="crispEdges" x="0" y="0" width="300" height="600" fill="url(#Gradient1)"/>
+				<rect shape-rendering="crispEdges"  x="300" y="0" width="300" height="600" fill="url(#Gradient2)"/>
+			</g>
+			</pattern>
+		</defs>
+		<path class="rotating" style="stroke: url(#Pattern);" fill='transparent' stroke-width='80' d='M 364 58 A 250 250 0 1 1 235 58'/>
+	</svg>
```
Comment (majidsajadi) [inline src/spinner/spinner.ts:35]: I suggest not using SVG for this component and going for a more CSS-centric approach. 
some thing like  [this](https://cssloaders.github.io/).
this way we have more flexibility to animate the spinner and/or adding new spinners if needed

File: src/spinner/index.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+* @summary Display spinner.
+*
+* @prop {string} [variant='primary'] - Specifies the spinner color. Acceptable values are `primary` and `inverse`.
+* 
+* @csspart svg - The svg tag that represents spinner component.
+* 
+* @cssprop [--tap-spinner-color-primary=--tap-sys-color-surface-black]
+* @cssprop [--tap-spinner-color-inverse=--tap-sys-color-surface-white]
+* @cssprop [--tap-spinner-size=--tap-sys-spacing-8]
+* @cssprop [--tap-spinner-padding=--tap-sys-spacing-2]
```
Comment (amir78729) [inline src/spinner/index.ts:30]: > [!TIP]
> Wrapping variables between backticks (`) in JSDoc will make the documentation more beautiful.  💅

```suggestion
* @prop {`string`} [`variant`=`'primary'`] - Specifies the spinner color. Acceptable values are `primary` and `inverse`.
*
* @csspart `svg` - The svg tag that represents spinner component.
*
* @cssprop [`--tap-spinner-color-primary`=`--tap-sys-color-surface-black`]
* @cssprop [`--tap-spinner-color-inverse`=`--tap-sys-color-surface-white`]
* @cssprop [`--tap-spinner-size`=`--tap-sys-spacing-8`]
* @cssprop [`--tap-spinner-padding`=`--tap-sys-spacing-2`]
```

File: src/spinner/spinner.stories.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import { html, TemplateResult } from "lit";
```
Comment (amir78729) [inline src/spinner/spinner.stories.ts:1]: Suggestion: I think the eslint is set up in our project and you can format your codes in these changes.

**Comment by amir78729** [comment]
@mahdieabdollahian well done, thanks 🍾 

close: #29

**Pull request merged**